sourcecode/chesssimul/board.py

```diff
@@ -1,169 +1,182 @@
-        board_string = ''
-        for line in range(8):
-            board_string += '|'
-            for col in range(8):
-                square = self.board[line][col]
-                if square.isEmpty():
-                    board_string += '  '
-                else:
-                    board_string += square.piece.to_unicode()
-                board_string += '|'
-            board_string += '\n'
-        return board_string
-
-    def move(self, lam: str):
-        lamlist = list(lam)
-        pawn_passant = ''
-        start_square: Square = self.board[Lines.get(lam[1])][Columns.get(lam[0])]
-        arrival_square: Square = self.board[Lines.get(lam[3])][Columns.get(lam[2])]
-        arrival_square.piece = start_square.piece
-        start_square.piece = None
-
-    def to_fen(self) -> str:
-        board_fen = ''
-        for line in range(8):
-            board_fen += '/'
-            empty_squares = 0
-            for col in range(8):
-                square = self.board[line][col]
-                if square.isEmpty():
-                    empty_squares += 1
-                    board_fen += ''
-                else:
-                    if empty_squares > 0:
-                        empty_number = str(empty_squares)
-                        board_fen += empty_number
-                    piece = str(square.piece.piece_to_str())
-                    board_fen += piece
-                    empty_squares = 0
-            if empty_squares > 0:
-                empty_number = str(empty_squares)
-                board_fen += empty_number
-        board_fen = board_fen[1:]
-        if self.to_move == Color.BLACK:
-            board_fen += " b "
-        else:
-            board_fen += " w "
-
-        if self.can_white_king_side_castle:
-            board_fen += "K"
-        if self.can_white_queen_side_castle:
-            board_fen += "Q"
-        if self.can_black_king_side_castle:
-            board_fen += "k"
-        if self.can_black_queen_side_castle:
-            board_fen += "q"
-
-        if self.en_passant_target_square is not None:
-            enpassant = self.en_passant_target_square
-            chessline = IndexToLine[enpassant[0]]
-            chesscolumn = IndexToColumn[enpassant[1]]
-            enpassant_target = ' ' + chessline + chesscolumn + ' '
-            board_fen += enpassant_target
-        else:
-            board_fen += ' - '
-
-        halfmove = str(self.halfmove_clock)
-        board_fen += halfmove
-        board_fen += ' '
-        fullmove = str(self.fullmove_number)
-        board_fen += fullmove
-
-        return board_fen
-
-
-board: Board = Board()
-board.init_board()
-print(board.to_unicode())
-board.move('e2e4')
-board.setEnPassantTargetSquare()
-print(board.to_unicode())
-print(board.to_fen())
-print(board.board[0][0].piece.kind)
-print(board.board[0][0].piece.color)
-print(board.board[4][4].piece.kind)
-print(board.board[4][4].piece.color)
-
+#from sourcecode.chesssimul.piece import Piece

```

Comment:
It seems that your Pull Request overrides the changes made by Olivia's pull request (deletion of the to_fen function inside the board class). You should pull the changes on the master branch and then do a merge on your own branch in order to prevent this overriding from happening.